src/js/pages/Topic.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react';
+import { get } from 'lodash';
+import axios from 'axios';
+
+// TODO: this jsonFileData is temporary. Add it to Wagtail API
+import jsonFileData from '__tmpdata/services';
+import FormFeedback from 'js/page_sections/FormFeedback';
+import Service311 from 'js/page_sections/Service311';
+import ListLink from 'js/modules/ListLink';
+import topicPageQuery from 'js/queries/topicPageQuery';
+
+class Topic extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: {}
+    };
+
+  }
+
+  componentDidMount() {
+    this.fetchData(this.props.match.params.id);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // only refetch data when props have changed
+    // this happens only when the route is updated
+
+    if (nextProps.match.params.id !== this.props.match.params.id) {
+      this.fetchData(nextProps.match.params.id);
+    }
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    // only rerender component when state has changed
+    // this happens only when data is refetched
+    if (nextState.data.id !== this.state.data.id) return true;
+
+    return false;
+  }
+
+  fetchData(id) {
```

Comment:
We're using this method in a couple different Page Components. And it seems like the `componentDidMount` and `shouldComponentUpdate` patterns are pretty consistent too. Not saying now is the time, but should we maybe think about a uniform abstraction for fetching data from the GraphQL server?

Reply:
My first instinct was a utility function, but i'd be open to seeing how other React-y patterns my be useful here.